### PR TITLE
Added a configmap to specify the two config options of mbs-backend.

### PR DESCRIPTION
**Pull request by AdamSaleh**
Based on the https://github.com/fedora-infra/mbbox/blob/master/components/mbs_backend/configmap.yml and the work of @odra on the image, I assume that only thing belonging to this config-map are the two config values to be injected as env-vars.

The start/entrypoint clearly belong in the image itself, while certs would be created as secrets and mounted later during deployment.

Wdyt?

### CODE REVIEW

**Comment by Zlopez** [comment]
I think these two values will go away eventually when switching to fedora messaging.

**Comment by Zlopez** [comment]
And you should also look in the `mbs_shared` folder. There will be probably be another configuration values.

**Comment by odra** [comment]
I don't think we need a configmap for this, we can just just use those values (from the cr) directly into the deployment as env vars.

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbmbsbackend_cr.yaml [hunk 1]
```diff
@@ -3,5 +3,6 @@ kind: MBMbsBackend
 metadata:
   name: example-mbmbsbackend
 spec:
-  # Add fields here
+  mbs_backend_consumer: true
+  mbs_backend_poller: true
   size: 3
```
Comment (odra) [inline mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbmbsbackend_cr.yaml:8]: this can be removed, it is auto generated by the sdk cli.

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,194 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ mbs_backend_configmap }}
+  labels:
+    app: mbs_backend
```
Comment (odra) [inline mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:6]: can you use "mbs-backend" as we are using in other components (koji-hub, koji-builder, etc).

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+          commit: f29
+          requires: {}
+          koji_tag: module-f29-build
+          mse: TRUE
+    document: modulemd
+    version: 1
+
+  koji.conf: |-
+    [koji]
+    server = https://koji-hub:8443/kojihub
```
Comment (odra) [inline mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:38]: can you use a cr variable "mbs_hub_host" that defaults to "koji-hub:8443"?

we do the same for other components

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 2 above)
Comment (AdamSaleh) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:6]: so, we use underscores only when referencing filesystem files, but dashes when configs/key-words?

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 3 above)
Comment (AdamSaleh) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:38]: ah, didn't notice this one, thanks!

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        HOST = '0.0.0.0'
+        PORT = 5000
+
+        # Global network-related values, in seconds
+        NET_TIMEOUT = 120
+        NET_RETRY_INTERVAL = 30
+
+        SYSTEM = 'koji'
+        MESSAGING = 'fedmsg'  # or amq
+        MESSAGING_TOPIC_PREFIX = ['org.fedoraproject.dev']
```
Comment (odra) [inline mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:74]: I think this could be a cr param

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        # Where we should run when running "manage.py runssl" directly.
+        HOST = '0.0.0.0'
+        PORT = 5000
+
+        # Global network-related values, in seconds
+        NET_TIMEOUT = 120
+        NET_RETRY_INTERVAL = 30
+
+        SYSTEM = 'koji'
+        MESSAGING = 'fedmsg'  # or amq
```
Comment (odra) [inline mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:73]: could be a cr param

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        POLLING_INTERVAL = 600
+
+        # Determines how many builds that can be submitted to the builder
+        # and be in the build state at a time. Set this to 0 for no restrictions
+        # New name
+        NUM_CONCURRENT_BUILDS = 5
+        # Old name https://pagure.io/fm-orchestrator/issue/574
+        NUM_CONSECUTIVE_BUILDS = 5
+
+        RPMS_DEFAULT_REPOSITORY = 'git+https://src.fedoraproject.org/rpms/'
```
Comment (odra) [inline mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:96]: @Zlopez is there value in adding those url values as cr params?

If we want/need to point it to other urls that are not prod environments.

Comment (AdamSaleh) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:96]: hm, abstracting these is not a problem

Comment (Zlopez) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:96]: @odra Yes, there is if you want to work with staging environment for testing.

File: mbox-operator/roles/mbs-backend/defaults/main.yml [hunk 7]
```diff
@@ -1,2 +1,3 @@
 ---
 # defaults file for mbs-backend
+mbs_backend_configmap: "{{ configmap | default('mbs-backend-configmap') }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/defaults/main.yml:3]: This should be a CR param `configmap`.

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 [hunk 8]
```diff
@@ -0,0 +1,194 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ mbs_backend_configmap }}
+  labels:
+    app: mbs_backend
+data:
+  default_module_f29.yml: |-
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:8]: I think there will be a new config file with each distribution release. @odra How do you think we should approach this?

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 4 above)
Comment (Zlopez) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:74]: I think we should have plenty of those configurable. We should allow user to just use our configmap as a template and create their own.

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 2 above)
Comment (odra) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:6]: in kubernetes config yes, usually "_" for ansible variables and etc.

**Comment by Zlopez** [comment]
As I see it, we have two options here:
1) Make most of the options configurable through CR
2) Provide a config template and just allow user to use their own configmap (this should be allowed anyway)

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 6 above)
Comment (odra) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:96]: I think it might be a good idea to create default vars for each url then.

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 8 above)
Comment (odra) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:8]: we will probably need a new operator release using the proper image tags for each fedora release

Comment (odra) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:8]: also looks like we could use a "version array" property to generate those yaml files in a loop

**Comment by odra** [comment]
I am wondering if the configmap should only mount yaml files and config.py be part of the container image using env vars for those variable values.. what are your thoughts on this?

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 4 above)
Comment (AdamSaleh) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2]: Made this CR param

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 6 above)
Comment (AdamSaleh) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2]: Didn't make default value, but CR var.

File: mbox-operator/roles/mbs-backend/defaults/main.yml (hunk 7 above)
Comment (AdamSaleh) [reply mbox-operator/roles/mbs-backend/defaults/main.yml:3]: Ok

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 8 above)
Comment (AdamSaleh) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:8]: Ok, didn't change anything here.

**Comment by AdamSaleh** [comment]
@odra I think having config.py in image and over-riding that with env-vars is better than mounting it.

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbmbsbackend_cr.yaml [hunk 9]
```diff
@@ -1,8 +1,22 @@
 apiVersion: apps.fedoraproject.org/v1alpha1
 kind: MBMbsBackend
 metadata:
-  name: example-mbmbsbackend
+  name: example-mbmbs-backend
```
Comment (Zlopez) [inline mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbmbsbackend_cr.yaml:4]:
```suggestion
  name: mb-mbs-backend
  label:
    app: mb-mbs-backend
```

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbmbsbackend_cr.yaml [hunk 10]
```diff
@@ -1,8 +1,22 @@
...
+  consumer: true
+  poller: true
+  config_system: 'koji'
+  config_messaging: 'fedmsg'
+  config_topic_prefix: 'org.fedoraproject.dev'
+  config_scm_url: 'git+https://src.fedoraproject.org/modules/'
+  config_rpms_def_repo: 'git+https://src.fedoraproject.org/rpms/'
+  config_rpms_def_cache: 'https://src.fedoraproject.org/repo/pkgs/'
+  config_modules_def_repo: 'git+https://src.fedoraproject.org/modules/'
+  config_koji_repo_url: 'https://kojipkgs.stg.fedoraproject.org/repos'
```
Comment (Zlopez) [inline mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbmbsbackend_cr.yaml:21]: Do we have staging values here for a reason?

I see some of the urls are pointing to staging and other don't.

File: mbox-operator/roles/mbs-backend/templates/configmap.yml.j2 (hunk 8 above)
Comment (Zlopez) [reply mbox-operator/roles/mbs-backend/templates/configmap.yml.j2:8]: The users can use their own configmap, so we could just left this to them. Just provide them with the template to use for creating their own.

**Comment by AdamSaleh** [comment]
recheck

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbmbsbackend_cr.yaml (hunk 10 above)
Comment (AdamSaleh) [reply mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbmbsbackend_cr.yaml:21]: I have no idea, I just took the values from https://github.com/fedora-infra/mbbox/blob/master/components/mbs_shared/configmap.yml as is.

Comment (Zlopez) [reply mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbmbsbackend_cr.yaml:21]: In this case, if they are configurable, it doesn't matter.

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbkojihub_cr.yaml [hunk 11]
```diff
@@ -8,7 +8,7 @@ spec:
   image: quay.io/fedora/koji-hub:latest
   replicas: 1
   persistent: true
-  host: 'koji-hub'
+  host: 'koji-hub:8443'
```
Comment (odra) [inline mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbkojihub_cr.yaml:11]: why did you change this value?

Comment (odra) [reply mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbkojihub_cr.yaml:11]: tests are failing because this value is used on other resources.. such as ingresses and and services.

Comment (AdamSaleh) [reply mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbkojihub_cr.yaml]: yep, took me a bit to figure that out

**Comment by Zlopez** [comment]
When I started working on the mbs_frontend_configmap, I came to a conclusion that it will be better to have two separate configmaps similar to original script. One shared between mbs-frontend and mbs-backend and the other separate. Looking at the deployment config there are even mounted to separate mount points. What do you think?

**Comment by odra** [comment]
I think we can do it as an improvement (new issue) and use one configmap for now if that make sense.

**Comment by Zlopez** [comment]
@odra You are working on the deployment part for mbs_backend. Does it make sense to have only one configmap? As I was looking at it there are two different mount points for configmaps.

**Comment by odra** [comment]
we can mount several volumes from one configmap by selecting a specific configmap key.

**Comment by Zlopez** [comment]
Could we do it the other way? I see that this PR splits the configmap to two separate ones, `configmap.yaml.j2` and `configmap_default_module.yaml.j2`. The `configmap_default_module.yaml.j2` must be in one mount point together with few pieces from `configmap.yaml.j2`.

**Comment by Zlopez** [comment]
I think the shared config for MBS is a good candidate for mbox shared data same as root ca.

**Comment by odra** [comment]
yeah I think it makes sense to have one configmap only

**Comment by Zlopez** [comment]
I would split it to two, but in the same way as the original script did this.

File: mbox-operator/roles/mbs-backend/templates/configmap_default_module.yml.j2 [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+      name: platform
+      profiles:
+        buildroot:
+          rpms: [bash, bzip2, coreutils, cpio, diffutils, fedora-release, findutils, gawk,
+            grep, gzip, info, make, patch, redhat-rpm-config, rpm-build, sed, shadow-utils,
+            tar, unzip, util-linux, which, xz]
+        srpm-buildroot:
+          rpms: [bash, fedora-release, fedpkg-minimal, gnupg2, redhat-rpm-config, rpm-build,
+            shadow-utils]
+      stream: f29
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/templates/configmap_default_module.yml.j2:22]:
```suggestion
      stream: f{{ mbs_backend_fedora_version }}
```

File: mbox-operator/roles/mbs-backend/templates/configmap_default_module.yml.j2 [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+      summary: Fedora {{mbs_backend_config_fedora_version}} traditional base
+      context: 00000000
+      version: 5
+      xmd:
+        mbs:
+          buildrequires: {}
+          commit: {{mbs_backend_config_fedora_version}}
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/templates/configmap_default_module.yml.j2:29]:
```suggestion
          commit: f{{mbs_backend_config_fedora_version}}
```

File: mbox-operator/roles/mbs-backend/templates/configmap_default_module.yml.j2 [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+          requires: {}
+          koji_tag: module-{{mbs_backend_config_fedora_version}}-build
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/templates/configmap_default_module.yml.j2:31]:
```suggestion
          koji_tag: module-f{{mbs_backend_config_fedora_version}}-build
```

File: mbox-operator/roles/mbs-backend/templates/configmap_default_module.yml.j2 [hunk 15]
```diff
@@ -0,0 +1,34 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ mbs_backend_configmap }}-{{mbs_backend_config_fedora_version}}
+  labels:
+    app: mbs-backend
+data:
+  default_module_{{mbs_backend_config_fedora_version}}.yml: |-
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/templates/configmap_default_module.yml.j2:8]:
```suggestion
  default_module_f{{mbs_backend_config_fedora_version}}.yml: |-
```

File: mbox-operator/roles/mbs-backend/defaults/main.yml [hunk 16]
```diff
@@ -1,5 +1,19 @@
...
+mbs_backend_config_fedora_version: "{{ fedora_version | default('f29') }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/defaults/main.yml:4]:
```suggestion
mbs_backend_config_fedora_version: "{{ fedora_version | default('32') }}"
```

**Comment by odra** [comment]
it can be merged after ci tests are green

**Comment by odra** [comment]
recheck
